src/Fixer/ClassNotation/ClassAttributesSeparationFixer.php

```diff
@@ -346,16 +418,53 @@ private function fixSpaceAboveClassElement(Tokens $tokens, int $classStartIndex,
             // there should be one linebreak between the element and the attribute above it
             $this->correctLineBreaks($tokens, $nonWhiteAbove, $firstElementAttributeIndex, 1);
 
-            // make sure there is blank line above the comment (with the exception when it is directly after a class opening)
-            $nonWhiteAbove = $this->findAttributeBlockStart($tokens, $nonWhiteAbove);
-            $nonWhiteAboveComment = $tokens->getNonWhitespaceSibling($nonWhiteAbove, -1);
+            // make sure there is blank line above the attribute comment
+            // with the exception when it is directly after a class opening or has an adjacent PHPDoc
+            list($nonWhiteAbove, $nonWhiteAboveComment, $reqLineCount) = $this->fixPossibleAttributePhpdocSequence($tokens, $nonWhiteAbove, $classStartIndex);
 
-            $this->correctLineBreaks($tokens, $nonWhiteAboveComment, $nonWhiteAbove, $nonWhiteAboveComment === $classStartIndex ? 1 : 2);
+            $this->correctLineBreaks($tokens, $nonWhiteAboveComment, $nonWhiteAbove, $reqLineCount);
 
             return;
         }
 
-        $this->correctLineBreaks($tokens, $nonWhiteAbove, $firstElementAttributeIndex, $nonWhiteAbove === $classStartIndex || self::SPACING_NONE === $spacing ? 1 : 2);
+        $reqLineCount = $this->determineRequiredLineCount($tokens, $nonWhiteAbove, $classStartIndex, $spacing);
+
+        $this->correctLineBreaks($tokens, $nonWhiteAbove, $firstElementAttributeIndex, $reqLineCount);
+    }
+
+    /**
+     * @return array<int, int>
+     */
+    private function fixPossibleAttributePhpdocSequence(Tokens $tokens, int $attributeCloseIndex, int $classStartIndex): array
```

Comment:
as the function doesn't actually fix anything I think it might be better called something like `getAttributePhpdocSequenceOffsetIndex` or something like that